Source/JavaScriptCore/runtime/JSGenericTypedArrayViewPrototype.cpp

```diff
@@ -31,6 +31,43 @@
 
 namespace JSC {
 
+JSC_DEFINE_HOST_FUNCTION(uint8ArrayPrototypeSetFromHex, (JSGlobalObject* globalObject, CallFrame* callFrame))
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    JSUint8Array* uint8Array = jsDynamicCast<JSUint8Array*>(callFrame->thisValue());
+    if (UNLIKELY(!uint8Array))
+        return throwVMTypeError(globalObject, scope, "Uint8Array.prototype.setFromHex requires that |this| be a Uint8Array"_s);
+
+    IdempotentArrayBufferByteLengthGetter<std::memory_order_seq_cst> byteLengthGetter;
+    if (UNLIKELY(isIntegerIndexedObjectOutOfBounds(uint8Array, byteLengthGetter)))
+        return throwVMTypeError(globalObject, scope, typedArrayBufferHasBeenDetachedErrorMessage);
+
+    JSString* jsString = jsDynamicCast<JSString*>(callFrame->argument(0));
+    if (UNLIKELY(!jsString))
+        return throwVMTypeError(globalObject, scope, "Uint8Array.prototype.setFromHex requires a string"_s);
+    if (UNLIKELY(jsString->length() % 2))
+        return JSValue::encode(throwSyntaxError(globalObject, scope, "Uint8Array.prototype.setFromHex requires a string of even length"_s));
+
+    StringView view = jsString->view(globalObject);
+
+    size_t read = 0;
+    size_t written = 0;
+    while (read < view.length() && written < uint8Array->length()) {
```

Comment:
We can compute how many iterations we need to do based on uint8Array size limit and view's length. Let's compute this and do iteration with that count.